### ISSUE TITLE
Implement License view

### ISSUE THREAD

**Issue by jvillalobos** [mozilla/addons-frontend#3697]
When you click on the license link on a listing page (like [this one](https://addons.mozilla.org/addon/onetab/)), the old site is shown. We should open a license page or show it inline. I suppose the same approach would apply for this and the privacy policy (#1828). /cc @pwalm for ux.

**Comment by bobsilverberg** [comment]
This is available via http://addons-server.readthedocs.io/en/latest/topics/api/addons.html#detail.

Pinging @pwalm for a mockup.

**Comment by pwalm** [comment]
EULA/License/Privacy Policy page:
![privacy policy page - 1366px](https://user-images.githubusercontent.com/6797635/43019682-a90125e8-8c12-11e8-9a03-db0870a29791.png)

**Comment by EnTeQuAk** [comment]
Cross-linking https://github.com/mozilla/addons/issues/765 as an fyi to test escaping of links on this page.

**Comment by bobsilverberg** [comment]
I have a few questions about this.

1. What is the relationship between the `licence`, as specified in the add-on's `version.license` property (which has properties for `name`, `text` and `url`) and the EULA, as specified in the add-on's `eula` property? Are these two different things, and if so, does each require a link on the add-on details page and a separate page for displaying the details for each?

2. If the `licence` contains a URL, the legacy site (I believe) just links to that URL, sending a user out of the site. Do we want to continue to do that, or do we want to render the contents of that URL in an iframe on a new "license" page?

3. What does the `text` property of the `version.license` property contain? I see that it is excluded from the regular add-on detail API response but is it something we need for a new "license" page, or can it be ignored?

I'm not sure who can answer the questions above, so soliciting feedback from @jvillalobos, @pwalm, @diox and anyone else who can help.

**Comment by bobsilverberg** [comment]
Actually, the add-on at https://addons.mozilla.org/en-US/firefox/addon/onetab/ seems to answer some of those questions. It looks like there can be 3 links for an add-on:
1. License, which comes from `version.license`.
2. Privacy Policy, which comes from `privacy_policy`.
3. End-User License Agreement, which comes from `eula`.

That still leaves a couple of questions:
1. I assume that the License page should show the text of a custom license if some exists, and that appears to be stored in `version.license.text`. Is that correct?
2. (Copied from above): If the `licence` contains a URL, the legacy site (I believe) just links to that URL, sending a user out of the site. Do we want to continue to do that, or do we want to render the contents of that URL in an iframe on a new "license" page?

**Comment by jvillalobos** [comment]
>  I assume that the License page should show the text of a custom license if some exists, and that appears to be stored in version.license.text. Is that correct?

The License (or EULA) is a completely separate field from the source code license. The link for the License should only appear when there's a License set, and shouldn't default to the code license.

> (Copied from above): If the licence contains a URL, the legacy site (I believe) just links to that URL, sending a user out of the site. Do we want to continue to do that, or do we want to render the contents of that URL in an iframe on a new "license" page?

I think we should render the URL in the License page, which I believe is what we do for the Privacy Policy field.

**Comment by bobsilverberg** [comment]
Note that the EULA portion of this will be addressed via PR #6833.

**Comment by bobsilverberg** [comment]
> Depending on how you are implementing the license on frontend maybe you want to change this behaviour?

We're going to render the text of the license, so I think it would be better to just know that a custom license exists. I'll open an issue for the API requesting that.

**Comment by bobsilverberg** [comment]
I've opened https://github.com/mozilla/addons-server/issues/9914 and https://github.com/mozilla/addons-server/issues/9915 to simplify implementing this page, so I'm going to mark this as blocked for now.

**Comment by bobsilverberg** [comment]
> I think we should render the URL in the License page, which I believe is what we do for the Privacy Policy field.

Sorry @jvillalobos, I shouldn't really have suggested that as an option because there are a number of issues with doing that, and currently we only display a link to the license URL, so this would be more of an enhancement rather than working towards parity and the ability to shut down the legacy pages.

For now we'll keep a non-custom license as just a link that points to the URL of the license. If we want to look at changing that in the future, so the contents of that URL renders inside an AMO page we can do that via a separate issue.

Does that seem okay?

**Comment by jvillalobos** [comment]
Sounds good.

**Comment by ioanarusiczki** [comment]
@bobsilverberg 
- I filed https://github.com/mozilla/addons-frontend/issues/7178 and https://github.com/mozilla/addons-frontend/issues/7179 for what I've noticed. Except those, page looks good, I tested on AMO dev with FF63(Win10).

- There's an issue about the HTML not being rendered in Privacy and EULA Pages https://github.com/mozilla/addons-frontend/issues/6897 but for the Custom License the text field (from Dev Hub) has no info about "Some HTML supported". 
I don't know if this is something that should be implemented too License too.
